src/models/event-timeline.ts

```diff
@@ -24,6 +24,30 @@ import { MatrixEvent } from "./event";
 import { Filter } from "../filter";
 import { EventType } from "../@types/event";
 
+export interface IInitialiseStateOptions {
+    /** Whether the timeline was empty before the marker arrived in
+     *  the room. This could be happen in a variety of cases:
+     *  1. From the initial sync
+     *  2. It's the first state we're seeing after joining the room
+     *  3. Or whether it's coming from `syncFromCache` */
+    timelineWasEmpty?: boolean;
+}
+
+export interface IAddEventOptions {
```

Comment:
This looks like a good base interface for the two lengthier ones in event-timeline-set

Reply:
More context in https://github.com/matrix-org/matrix-js-sdk/pull/2299#discussion_r883502814